woocommerce/woocommerce-gutenberg-products-block
# upgrade Storybook to 6.0

Fixes #3028 

Upgrading Storybook to latest. More info / migration guides:

- https://medium.com/storybookjs/storybook-6-migration-guide-200346241bb5
- https://github.com/storybookjs/storybook/blob/next/MIGRATION.md

Changes:

- Storybook packages all updated (via `npx sb upgrade`).
- Added a dependency on [`core-js`](https://github.com/zloirock/core-js). This is not strictly a dependency of Storybook but is [needed as a workaround](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#core-js-dependency-errors) (our story build fails without it).

There's more we can do - 6.0 has a bunch of cool new stuff (in particular: essentials + controls + docs). Saving that for other PRs since we don't need to do it to upgrade. Added a follow-up issue: https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/3041

### How to test the changes in this Pull Request:

1. Clone this branch. 
2. Install dependencies - `npm install` or `npm ci`.
3. Run storybook - `npm run storybook`.
4. Browse storybook (runs on port 6006 by default) and confirm nothing's broken.
